Create test suite

Coolroo opened the pull request:
Created VSC test suite & action to run it

Coolroo committed:
create test files

Coolroo committed:
Imported required libraries

Coolroo committed:
modified:   package.json

Coolroo committed:
added testing suite & added testing functions to parser

Coolroo committed:
Added test suite and added testing functions to parser

Coolroo committed:
Merge remote-tracking branch 'origin' into create-test-suite

Coolroo committed:
added test files

Coolroo committed:
added python testing

Coolroo committed:
Finished python tests

Coolroo committed:
finsihed java tests

Coolroo committed:
Finished Javascript tests

Coolroo committed:
Changed codeParser to return a promise in the constructor
updates tests to accommodate the above change
added additional launch to run extension

calum-bird reviewed (commented):
Two things:
1. Let's remove the yarn.error.log file and add it to .gitignore to prevent accidental addition in the future.
2. For each test suite, can you create some constant variables denoting the expected number of documented and undocumented functions? Right now they are hardcoded and it's unclear at a glance where to go to change them. Ideally they are just  at the top of the file like `EXPECTED_FUNCTION_COUNT=4`

Coolroo committed:
Updated tests to use constants
added yarn-error-log to .gitignore

Coolroo committed:
update version

calum-bird reviewed (approved):
As discussed, we'll expand these tests to make fewer assumptions at a later date. Looks good for barebones initial testing though!

Coolroo committed:
modified:   src/test/suite/parser/csharp.test.ts
	modified:   src/test/suite/parser/java.test.ts
	modified:   src/test/suite/parser/javascript.test.ts
	modified:   src/test/suite/parser/python.test.ts
